lisa/util/__init__.py

```diff
@@ -207,6 +207,21 @@ def __str__(self) -> str:
         return message
 
 
+class NotEnoughMemoryException(LisaException):
+    """
+    This exception is used to indicate that that the system does not have enough memory.
+    """
+
+    def __init__(self, message: str = "") -> None:
+        self._extended_message = message
+
+    def __str__(self) -> str:
+        message = "Not enough memory"
```

Comment:
Ideally this would be more descriptive, I'm picturing seeing a bunch of tests failing with "Not enough memory" as the only failure message. Providing an error msg like "Not enough memory for requested operation: Requested XXX, Available: YYY" would be more useful.

Reply:
_extended_message will take care of that

Reply:
Nice! You nailed it 😂